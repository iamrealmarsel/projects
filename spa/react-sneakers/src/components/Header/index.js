import { Link } from 'react-router-dom';
import { useCart } from '../../hooks/useCart';
import cn from './Header.module.scss';

function Header(props) {
  const { onCartClick } = props;
  const { totalPrice } = useCart();

  return (
    <header className={cn.header}>
      <Link to='/'>
        <div className={cn.headerLeft}>
          <img width='40' height='40' src='img/logo.png' alt='' />
          <div className={cn.headerInfo}>
            <h3 className={cn.headerInfoTitle}>React Sneackers</h3>
            <p className={cn.headerInfoDesc}>Магазин лучших кроссовок</p>
          </div>
        </div>
      </Link>
      <ul className={cn.headerRight}>
        <li onClick={onCartClick} className={cn.headerRightCart}>
          <img src='img/cart.svg' alt='' />
          <span>{totalPrice} руб.</span>
        </li>
        <li className={cn.headerRightFavorites}>
          <Link to='/favorites'>
            <img src='img/favorites.svg' alt='' />
          </Link>
        </li>
        <li className={cn.headerRightProfile}>
          <Link to='/orders'>
            <img src='img/user.svg' alt='' />
          </Link>
        </li>
      </ul>
    </header>
  );
}

export default Header;
