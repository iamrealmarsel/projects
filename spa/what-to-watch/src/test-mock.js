import {createStore} from 'redux';
import rootReducer from 'store/root-reducer';
import {loadMoviePromo} from 'store/actions/load';


export const noop = () => {};
export const VISIBLE_MOVIES_COUNT = 16;
export const ID = 1;
export const RATING_STARS = [false, false, true, false, false];
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const CURRENT_GENRE = `Comedy`;
export const ALL_GENRES = `All genres`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {params: {id: 1}};
export const PARAMS = {id: 1};

export const MOVIES = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    backgroundColor: `#73B39A`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    releaseYear: 2008,
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    backgroundColor: `#73B39A`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    releaseYear: 2008,
    myList: false,
    comments: false,
  },
];

export const MOVIE = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  preview: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  backgroundColor: `#73B39A`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  releaseYear: 2008,
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    id: 111,
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    author: `Christina`,
    message: `This film is an experience and i has already seen it 4 times`,
  },
  {
    id: 222,
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    author: `Christina`,
    message: `This film is an experience and i has already seen it 4 times`,
  }
];

const store = createStore(rootReducer);
store.dispatch(loadMoviePromo(MOVIE));
export {store};
