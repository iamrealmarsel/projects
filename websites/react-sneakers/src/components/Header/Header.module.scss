.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
  margin-bottom: 40px;
  border-bottom: 1px solid #eaeaea;

  a {
    text-decoration: none;
  }
}

.headerLeft {
  display: flex;
  align-items: center;

  img {
    margin-right: 15px;
  }
}

.headerInfoTitle {
  text-transform: uppercase;
  margin: 0;
}

.headerInfoDesc {
  margin: 0;
  opacity: 0.5;
}

.headerRight {
  display: flex;
  align-items: center;
}

.headerRightCart {
  display: flex;
  align-items: center;
  margin-right: 30px;
  cursor: pointer;

  img {
    margin-right: 15px;
  }
}

.headerRightProfile {
  cursor: pointer;
}

.headerRightFavorites {
  margin-right: 30px;
}
