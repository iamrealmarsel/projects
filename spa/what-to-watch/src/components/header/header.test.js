import React from "react";
import {BrowserRouter} from 'react-router-dom';
import TestRenderer from "react-test-renderer";
import Header from "./header";
import * as mock from "test-mock";


test(`render Header`, () => {
  const tree = TestRenderer
    .create(
        <BrowserRouter >
          <Header
            id={mock.ID}
            noAvatar={true}
            title={mock.TITLE}
            movieTitle={mock.TITLE}
          />
        </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
