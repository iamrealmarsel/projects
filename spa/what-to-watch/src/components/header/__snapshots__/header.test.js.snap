// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <nav
    className="breadcrumbs"
  >
    <ul
      className="breadcrumbs__list"
    >
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link"
          href="/films/1"
          onClick={[Function]}
        >
          Bohemian Rhapsody
        </a>
      </li>
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link"
        >
          Add review
        </a>
      </li>
    </ul>
  </nav>
  <h1
    className="page-title user-page__title"
  >
    Bohemian Rhapsody
  </h1>
</header>
`;
